bot/src/rpc_client_utils.rs

```diff
@@ -36,7 +36,12 @@ fn new_spinner_progress_bar() -> ProgressBar {
     progress_bar
 }
 
-fn new_tpu_client_with_retry(
+pub struct Clients {
```

Comment:
we _could_ take the encapsulation a step further. un-`pub` the members, impl `Deref` with `type Target = RpcClient;` add a ctor to take an TPC/TPU selector, then override `send_transaction()`.

pretty sure it'd be less plumbing :wink:

Reply:
Did that in [2bd8a4b](https://github.com/solana-foundation/stake-o-matic/pull/456/commits/2bd8a4b737b77dabf9b0f2503cd1e4a9aa8c47c2) -- I'm not a huge fan of `impl Deref` since it obfuscates the code using it, but I also don't feel that strongly about it.  Since the plumbing was already there, I just reused that. Let me know what you prefer